src/lifetimes.md

```diff
@@ -1,9 +1,15 @@
 # Lifetimes
 
 Rust enforces these rules through *lifetimes*. Lifetimes are effectively
-just names for scopes somewhere in the program. Each reference,
-and anything that contains a reference, is tagged with a lifetime specifying
-the scope it's valid for.
+just names for areas of code somewhere in the program (you could also look at
+them as time intervals, except that the beginning and end of the time is denoted
+by when a line of code executes). Each reference, and anything that contains a
+reference, is tagged with a lifetime specifying the paths of executions it's
+valid for.
```

Comment:
This paragraph is now a bit too much of a mess of metaphors. Suggested rewrite:

```markdown
Rust enforces these rules through *lifetimes*. Lifetimes are named
regions of code that a reference must be valid for. Those regions
may be fairly complex, as they correspond to paths of execution
in the program. There may even be holes in these paths of execution, 
as it's possible to invalidate a reference as long as it's reinitialized
before it's used again. Types which contain references (or pretend to)
may also be tagged with lifetimes so that Rust can prevent them from
being invalidated as well.
```

Reply:
Not the first time that happened to me, I guess my brain just works mostly on metaphors most of the time :-|. Anyway, adopting your version, thank you.